internal/controllers/catalog_test.go

```diff
@@ -0,0 +1,97 @@
+package controllers_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/operator-framework/operator-controller/internal/controllers"
+
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+	operatorsv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ = Describe("Operator Controller Catalog Reconciler", func() {
+	When("a catalog changes on cluster", func() {
```

Comment:
Based on the previous suggestion, WDYT about adding an extra layer of hierarchy under the `Describe("SetupWithManager")` block. This way it'll be setup a bit easer to handle more watches or other configurations we add in the future.

```
BeforeEach() // setup and start the controller+manager
When("Operator events occur")
    It("reconciles self") // do Operator create, update, delete
When("Catalog events occur")
    BeforeEach() // create a few operators
    It("reconciles all Operators") // do Catalog create, update, delete
    AfterEach() // delete the operators
AfterEach() // cancel the context to stop the manager
```

Reply:
I've been thinking about this, but do we want to test controller-runtime objects? Testing for reconcile on catalog events makes sense with catalogs since we're using a custom handler, but with operators, it feels like we're testing whether the manager works, that should be on controller-runtime.

Reply:
It isn't really testing controller-runtime. It's testing our specific watch setup. I agree that almost all of the code that _actually_ runs is controller-runtime code, but it's all triggered based on our setup of the manager.

I think there's pretty good long term value here from a regression standpoint, especially if we can get the framework setup while things are simple. Anecdotally, the rukpak watches are actually fairly complex due to the way its APIs reference provisioner classes. There has been discussion in the context of oeprator-controller to implement things like upgrade policies via referenced objects, so I think it is inevitable that our watches become more complex over time.

Reply:
I think @joelanford's suggestion here makes a lot of sense for testing that the watches we are setting up are getting triggered as expected. That being said, I don't think it should block this PR specifically unless @ankitathomas really felt like including it. At least the foundation is laid by this PR and we can always increase the robustness of our tests in follow ups